packages/cashier/src/stores/general-store.js

```diff
@@ -298,19 +295,10 @@ export default class GeneralStore extends BaseStore {
         }
     }
 
-    async getAdvertizerError() {
-        const advertiser_info = await this.WS.authorized.p2pAdvertiserInfo();
-        this.setP2pAdvertiserError(getPropertyValue(advertiser_info, ['error', 'code']));
-    }
-
-    setP2pAdvertiserError(value) {
-        this.p2p_advertiser_error = value;
-    }
-
     checkP2pStatus() {
-        const advertiser_error = this.p2p_advertiser_error;
-        const is_p2p_restricted = advertiser_error === 'RestrictedCountry' || advertiser_error === 'RestrictedCurrency';
-        this.setIsP2pVisible(!(is_p2p_restricted || this.root_store.client.is_virtual));
+        const website_status = this.WS.wait('website_status');
+        const is_p2p_disabled = website_status?.website_status?.p2p_config?.disabled;
```

Comment:
@carol-binary In this [card](https://redmine.deriv.cloud/issues/85474#Farzin_/_Task_-_Replace_the_use_of_cashier_store_in_other_packages_with_shared_hooks) one of the things we need to do is to extract this `is_p2p_enabled` from cashier store to a reusable hook in `@deriv/hooks`.

What’s your opinion? Do you think we should be able to move it out or are there any specific use cases that make it tied up with the cashier?

By the way it’s not related to your changes, I'm just asking to have a better understanding 😅🙇🏻

Reply:
I think it can be moved out to @deriv/hooks since it's mainly used to determine if P2P will be visible in the cashier tabs. From the P2P side, we don't even use it, we let Cashier handle the tabs and decide if P2P will be visible there. Hope this answers your question 😄

Reply:
Ah got it 😅 Thank you for your explanation Carol 🌹